RHEL6_7/system/pam/check

```diff
@@ -4,22 +4,57 @@
 
 #END GENERATED SECTION
 
-POSTUPGRADE_DIR="$VALUE_TMP_PREUPGRADE/postupgrade.d/pam"
+POSTUPGRADE_DIR="$POSTUPGRADE_DIR/pam"
+CLEANCONF_DIR="$VALUE_TMP_PREUPGRADE/cleanconf/etc/pam.d"
+
 if [[ ! -d "$POSTUPGRADE_DIR" ]]; then
     mkdir -p "$POSTUPGRADE_DIR"
 fi
+if [[ ! -d "$CLEANCONF_DIR" ]]; then
+    mkdir -p "$CLEANCONF_DIR"
+fi
+
 SCRIPT_NAME="postupgrade-pam.sh"
 POST_SCRIPT="postupgrade.d/$SCRIPT_NAME"
+
 cp -f $POST_SCRIPT $POSTUPGRADE_DIR/$SCRIPT_NAME
+
 fail=0
 
+comment_invalid () {
+    perl -p -i -e 's/^/#/g if /^(?!.*#)(?=.*(pam_passwdqc|pam_ecryptfs))/;' "$1"
+}
+get_deprec_dirctv () {
+
+    deprec_dirctv="$1"
+    file="$2"
+    export deprec_dirctv
+    export file 
+    deprec_status=$(perl -ne '$deprec_dirctv = $ENV{deprec_dirctv};$file = $ENV{file}; 
+    print "PAM: The $file file contains the $deprec_dirctv module, which is no longer supported.\n"
+    if /^(?!.*#)(?=.*$deprec_dirctv)/' "$2")
```

Comment:
You should use everywhere the `$file` variable  or everywhere the `$2` variable. Do not mix them. In addition append indentation for inner code to improve readability. But much rather I would see deprec_status omitted completely for use outside of the scope of the function. You can just examine whether the deprecated directive is use in the given file or not and base on that print the output (that you can store outside of this function). IOW, the `deprec_status` should not be used here, but outside of the function.

Reply:
Fixed, I'm using renamed "$input" variable on all places. Not mixing anymore